### PR TITLE
Fix vault module, add new test with java client v2

### PR DESCRIPTION
**Pull request by casz**
resolves #1410

Using JSON and removing all the whitespace is one fix.

Perhaps we should add a second method in the that will call CLI using format=JSON and remove the whitespace 😓

### CODE REVIEW

**Comment by rnorth** [comment]
@casz sorry for the late posting of comments. Would you still be willing to push this PR through?

File: modules/vault/build.gradle [hunk 1]
```diff
@@ -3,5 +3,8 @@ description = "Testcontainers :: Vault"
 dependencies {
     compile project(':testcontainers')
 
-    testCompile 'io.rest-assured:rest-assured:4.0.0'
+    testCompile 'com.bettercloud:vault-java-driver:4.0.0'
+    testCompile 'io.rest-assured:rest-assured:3.3.0'
```
[the remark this answers is not in the document]
Comment (casz) [reply modules/vault/build.gradle]: I assume merge conflict 😅 I just tried to fix the initial PR

File: modules/vault/src/main/java/org/testcontainers/vault/VaultContainer.java [hunk 2]
```diff
@@ -18,33 +18,33 @@
...
+
     public VaultContainer() {
-        this("vault:0.7.0");
+        this("vault:1.1.3");
     }
 
     public VaultContainer(String dockerImageName) {
         super(dockerImageName);
+
+        setWaitStrategy(Wait.forHttp("/v1/secret/is_alive").forStatusCode(400));
```
[the remark this answers is not in the document]
Comment (rnorth) [reply modules/vault/src/main/java/org/testcontainers/vault/VaultContainer.java]: Wouldn't it make sense for this to be:
```
setWaitStrategy(Wait.forHttp("/v1/sys/health").forStatusCode(200));
```
(looking at https://www.vaultproject.io/api/system/health.html)

**Comment by rnorth** [comment]
I took the liberty of updating the PR to reflect my comment - I hope this is OK.
This seems to be a reliable way to test the container readiness, as far as I can tell.

@bsideup, @kiview would one of you mind sanity checking my change?

Additionally I noticed a couple of little things that are being flagged by the IDE - but _not_ related to this PR at all. I'll raise a separate PR to address those.

**Comment by casz** [comment]
All good @rnorth THANKS! sorry for not getting back to you 😭

**Comment by rnorth** [comment]
No probs @casz!

**Comment by casz** [comment]
LGTM 👍

**Comment by rnorth** [comment]
Thank you @casz and @robfrank for your work on this! Finally merged 😅

**Pull request merged**